### PR TITLE
#5400 water heater heat pump pumped condenser part load ratio calculation failed warning

### PR DESCRIPTION
**Pull request by rraustad**
## Pull request overview

Please change this line to a description of the pull request, with useful supporting information including whether it is a new feature, or fixes a defect, a cross reference to any defects addressed in this PR, the type of changes to be made, and whether diffs are expected/justified based on this change.
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
  - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
  - NewFeature: This pull request includes code to add a new feature to EnergyPlus
  - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
  - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog
### Review Checklist

This will not be exhaustively relevant to every PR.
- [ ] Code style (parentheses padding, variable names)
- [ ] Functional code review (it has to work!)
- [ ] If defect, results of running current develop vs this branch should exhibit the fix
- [ ] CI status: all green or justified
- [ ] Performance: CI Linux results include performance check -- verify this
- [ ] Unit Test(s)
- C++ checks:
  - [ ] Argument types
  - [ ] If any virtual classes, ensure virtual destructor included, other things
- IDD changes:
  - [ ] Verify naming conventions and styles, memos and notes and defaults
  - [ ] Open windows IDF Editor with modified IDD to check for errors
  - [ ] If transition, add rules to spreadsheet
  - [ ] If transition, add transition source
  - [ ] If transition, update idfs
- [ ] If new idf included, locally check the err file and other outputs
- [ ] Documentation changes in place
- [ ] ExpandObjects changes??
- [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: src/EnergyPlus/WaterThermalTanks.cc [hunk 1]
```diff
@@ -7468,6 +7469,103 @@ namespace WaterThermalTanks {
 		if ( HeatPump.Mode == HeatMode ) {
 			// HPWH was heating last iteration and will continue to heat until the set point is reached
 			HPPartLoadRatio = 1.0;
+			if ( TankTemp > SetPointTemp ) { // tank set point temp may have been reduced since last iteration and float mode may be needed
+				HeatPump.Mode = FloatMode;
```
Comment (rraustad) [inline src/EnergyPlus/WaterThermalTanks.cc:7569]: Note all this logic was added to HeatMode (whereas before only PLR=1 was used) to account for times when either the SP was reduced or the use side nodes could meet the set point. This corrects the defect file.

**Comment by nrel-bot-3** [comment]
@rraustad @lgentile it has been 14 days since this pull request was last updated.

**Comment by nmerket** [comment]
@rraustad @Myoldmopar I can review this.

**Comment by Myoldmopar** [comment]
Should we make a followup issue to address why the example file has the iteration failure?  I only noticed since it has ERR diffs.

**Comment by rraustad** [comment]
Yes, these models should not show these type of warnings. There is usually an underlying issue causing them that developers initially missed.

**Comment by Myoldmopar** [comment]
@rraustad I pushed some commits to clean up a build warning and the merge conflict.  The merge conflict was simply that two new unit tests got added to the same line of the unit test file, so resolving it was just to add both of them.  It was a little goofy because Git tried to match up the two different unit tests because some lines were common, so it looked like there were >20 conflicts, intermingled between a few common lines, but it was really just one big one conflict. 

I have nothing further here, so when CI is done, if it's fine, it'll merge.

**Comment by Myoldmopar** [comment]
So, this is fine, and I want to click merge, but I'm going to mark it as ready-to-merge in my checklist, and merge it later tonight (or early tomorrow morning).  That way it won't interfere with the other activity going on.

**Pull request merged**